src/Sarif.Viewer.VisualStudio/ErrorListService.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using Microsoft.CodeAnalysis.Sarif.Driver;
+using Microsoft.CodeAnalysis.Sarif.Driver.Sdk;
+using Microsoft.CodeAnalysis.Sarif.Sdk;
+
+namespace SarifViewer
+{
+    public class ErrorListService
+    {
+        public static ErrorListService Instance = new ErrorListService();
+
+        public static void ProcessSarifLog(ResultLog resultLog)
+        {
+            foreach (RunLog runLog in resultLog.RunLogs)
+            {
+                Instance.WriteRunLogToErrorList(runLog);
+            }
+
+            SarifTableDataSource.Instance.BringToFront();
+        }
+
+        private ErrorListService()
+        {
+            this.documentToLineIndexMap = new Dictionary<string, NewLineIndex>();
+        }
+
+        private Dictionary<string, NewLineIndex> documentToLineIndexMap;
+
+        private IRuleDescriptor GetRule(RunLog runLog, string ruleId)
+        {
+            if (runLog.RuleInfo == null)
+            {
+                return null;
+            }
+
+            foreach (IRuleDescriptor ruleDescriptor in runLog.RuleInfo)
+            {
+                if (ruleDescriptor.Id == ruleId) { return ruleDescriptor; }
+            }
+
+            throw new InvalidOperationException();
+        }
+
+        private void WriteRunLogToErrorList(RunLog runLog)
+        {
+            List<SarifError> sarifErrors = new List<SarifError>();
+
+            // Prefer optional fullName,  fall back to required Name property
+            string toolName = runLog.ToolInfo.FullName ?? runLog.ToolInfo.Name;
+
+            foreach (Result result in runLog.Results)
+            {
+                string message;
+                string document;
+                Region region;
+                NewLineIndex newLineIndex;
+
+                foreach (Location location in result.Locations)
+                {
+                    PhysicalLocationComponent physicalLocation;
+                    if (location.ResultFile != null)
+                    {
+                        physicalLocation = location.ResultFile[0];
+                    }
+                    else
+                    {
+                        physicalLocation = location.AnalysisTarget[0];
+                    }
+
+                    // TODO helper to provide best representation of URI
+                    document = physicalLocation.Uri.LocalPath;
+
+                    // TODO retrieve file from nested component
+                    region = physicalLocation.Region;
+
+                    IRuleDescriptor rule = GetRule(runLog, result.RuleId);
+                    message = result.GetMessageText(rule, concise: true);
```

Comment:
Suppose the optional `runLog.ruleInfo` property is absent, so that `GetRule` returns `null`, and suppose that both the `ShortMessage` and `FullMessage` properties of the `result` object are missing (I know that's illegal). Then I see that the method `GetMessageText` will assert.

Similar to above, I don't think this is the time to be doing that sort of constraint checking. And on a retail build where `Assert` is compiled out, `GetMessageText` will null-ref.

This is all fine as long as there is some code somewhere -- or a work item to write the code -- that does all the constraint validation.